File: backend/src/main/java/com/interviewcoach/common/api/CandidateProfileDto.java
package com.interviewcoach.common.api;

import java.util.List;

/**
 * 已确认的候选人画像 DTO，用户确认摘要后持久化到远端。
 *
 * @param id          画像 ID
 * @param targetId    关联的岗位目标 ID
 * @param summary     用户确认后的经历摘要
 * @param skills      技能列表
 * @param projects    项目经历列表
 * @param experience  工作经历列表
 * @param confirmedAt 确认时间
 */
public record CandidateProfileDto(String id, String targetId, String summary, List<String> skills, List<String> projects, List<String> experience, String confirmedAt) {
}

File: ios/InterviewCoach/InterviewCoach/Core/API/DTO/CandidateProfileDTO.swift
import Foundation

struct CandidateProfileDraftRequestDTO: Encodable {
  let resumeText: String?
  let projectRawText: String?
}

struct CandidateProfileDraftDTO: Decodable, Equatable {
  let summary: String
  let skills: [String]
  let projects: [String]
  let experience: [String]
  let rawTextLength: Int
}

struct CandidateProfileConfirmRequestDTO: Encodable {
  let targetId: String
  let summary: String
  let skills: [String]
  let projects: [String]
  let experience: [String]
}

struct CandidateProfileDTO: Decodable, Equatable {
  let id: String
  let targetId: String
  let summary: String
  let skills: [String]
  let projects: [String]
  let experience: [String]
  let confirmedAt: String
}
